src/twisted/internet/abstract.py

```diff
@@ -26,13 +28,15 @@ def _concatenate(bObj, offset, bArray):
         # in the result.
         return buffer(bObj, offset) + b"".join(bArray)
 except ImportError:
-    # Python 3
-    def _concatenate(bObj, offset, bArray):
-        # Python 3 lacks the buffer() builtin and the other primitives don't
-        # help in this case.  Just do the copy.  Perhaps later these buffers can
-        # be joined and FileDescriptor can use writev().  Or perhaps bytearrays
-        # would help.
-        return bObj[offset:] + b"".join(bArray)
+    if (sys.version_info.major, sys.version_info.minor) == (3, 3):
```

Comment:
Can we just change everything to do version checking instead of a combination of version checking and import probing?